version/VERSION

```diff
@@ -1 +1 @@
-0.14.0
+0.14.1
```

Comment:
For main, I think this should be `0.15.0` now?

Reply:
I believe this is used to determine the version of any artifact that is built on commits to a branch. So for example the artifacts from [this](https://github.com/hashicorp/boundary/commit/7d57a99e348965ec0b5403190329f997c7a3dfbb) commit [on main](https://github.com/hashicorp/boundary/actions/runs/6501744981) show as `0.14.0`.

I think this is supposed to be set to the *next* version so that the artifacts are already built with the correct version when we go to do a release. And since any 0.14.x version should come from the release/0.14.x branch, main should be set to 0.15.0.

Reply:
Interesting! I've made this change now: 

In the release branch, however, it wouldn't be set to `0.15.0` right?

Reply:
In the release branch it should be 0.14.1. And after 0.14.1 is released we should bump the release branch to 0.14.2, although that might happen automatically, I can't remember if that is part of the release automation now or not.

Reply:
Agreed with Tim. It also helps to ensure we're developing on `main` with the current deprecations. When we did `0.14.0`, we didn't bump the version number until release time and a bunch of tests started failing because of some deprecated features.